omod/src/main/webapp/pages/visit/templates/visitDetailsEdit.gsp

```diff
@@ -18,11 +18,11 @@
                 <date-with-popup ng-model="newStartDatetime" min-date="startDateLowerLimit" max-date="(startDateUpperLimit && (startDateUpperLimit < newStopDatetime) ? startDateUpperLimit : newStopDatetime) || now"></date-with-popup>
             </td>
         </tr>
-        <tr ng-show="newStopDatetime" >
+        <tr>
             <td>${ ui.message("uicommons.end") }</td>
             <td>{{ visit.stopDatetime | serverDate }}</td>
             <td>
-                <date-with-popup type="text" size="20" ng-model="newStopDatetime" min-date="(endDateLowerLimit && (endDateLowerLimit > newStartDatetime) ? endDateLowerLimit : newStartDatetime)" max-date="endDateUpperLimit || now"></date-with-popup>
+                <date-with-popup type="text" size="20" ng-model="newStopDatetime" min-date="(endDateLowerLimit && (endDateLowerLimit > newStartDatetime) ? endDateLowerLimit : newStartDatetime)" max-date="endDateUpperLimit || now" clear-button="endDateUpperLimit ? false: true"></date-with-popup>
```

Comment:
If this is JavaScript you can write
```suggestion
                <date-with-popup type="text" size="20" ng-model="newStopDatetime" min-date="(endDateLowerLimit && (endDateLowerLimit > newStartDatetime) ? endDateLowerLimit : newStartDatetime)" max-date="endDateUpperLimit || now" clear-button="!endDateUpperLimit"></date-with-popup>
```

But honestly I can't tell whether it's Angular or GSP or what :sweat_smile:

Reply:
It's not GSP.  Originally I had the ! format, but it wasn't working so I changed it just in case that was issue even though it turned out to be something else.  You are right, the ! should work, I'll change back.